.github/workflows/build_and_test.yml

```diff
@@ -379,6 +379,7 @@ jobs:
       SKIP_PACKAGING: true
       METASPACE_SIZE: 1g
       BRANCH: ${{ inputs.branch }}
+      SPARK_ANSI_SQL_MODE: true
```

Comment:
Check ansi is true , will revert this file after the test passes.

Reply:
https://github.com/LuciferYang/spark/runs/19046429659

![image](https://github.com/apache/spark/assets/1475305/eb090088-b04b-4e2b-aa21-b1eedaed6cd4)


SPARK_ANSI_SQL_MODE check passed